content/docs/esc/environments/configuring-oidc/doppler.md

```diff
@@ -0,0 +1,182 @@
+---
+title_tag: Configure OpenID Connect for Doppler | Pulumi ESC
+meta_desc: This page describes how to configure OIDC token exchange in Doppler for use with Pulumi
+title: Doppler
+h1: Configuring OpenID Connect for Doppler
+meta_image: /images/docs/meta-images/docs-meta.png
+menu:
+  esc:
+    name: Doppler
+    parent: esc-configuring-oidc
+    weight: 4
+---
+
+This document outlines the steps required to configure Pulumi to use OpenID Connect to authenticate with Doppler. OIDC
+in Doppler uses [service account identities](https://docs.doppler.com/docs/service-account-identities) to access
+Doppler resources. Access to the temporary credentials is authorized using identities that validate the contents of
+the OIDC token issued by the Pulumi Cloud.
+
+## Prerequisites
+
+* Your Doppler workplace must be on the Team or Enterprise plan in order to use service account identities.
+* You must be an admin of your Doppler workplace to create and configure service account identities.
+
+{{< notes type="warning" >}}
+Please note that this guide provides step-by-step instructions based on the official provider documentation which is
+subject to change. For the most current and precise information, always refer to
+the [official Doppler documentation](https://docs.doppler.com/docs/service-account-identities).
+{{< /notes >}}
+
+## Creating a Service Account
+
+To create a new service account, in the navigation pane of the [Doppler dashboard](https://dashboard.doppler.com):
+
+1. Select **Team**, **Service Accounts** and then click **New Service Account**.
+2. Provide a name for your service account (ex: `pulumi-esc-oidc-app`).
+3. Click **Create**.
+4. Select a **Role** for the service account or manually grant project access.
+
+## Add an Identity (OIDC Authentication)
+
+To add an identity to a service account:
+
+1. Navigate to the service account from **Team**, **Service Accounts**
+2. Click **New Identity**.
+3. Fill in the form fields as follows:
+    * **Discovery URL:** `https://api.pulumi.com/oidc`.
+    * **Audience:** This is different between Pulumi deployments and ESC. For Deployments this is only the name of your Pulumi organization. For ESC this is the name of your Pulumi organization prefixed with `doppler:` (e.g. `doppler:{org}`).
+    * **Subject:** `pulumi:environments:org:<your-pulumi-org>:env:<your-project>/<your-environment>` (see more examples at the end of this section).
+4. Click **Create Identity**.
+{{< notes type="info" >}}
+For environments in the `default` project the audience will use just the Pulumi organization name. This is to
+prevent regressions for legacy environments.
+{{< /notes >}}
+
+After the Identity has been created, take note of the following details:
+
+* Identity ID
+
```

Comment:
Spacing here is off ... but looks like a problem from our markdown converter

<img width="602" alt="image" src="https://github.com/user-attachments/assets/91c5e7b0-6b84-4d81-8d58-04f7d161099e" />

Reply:
Fixed for both Doppler and Infisical in db6b111